### PR TITLE
Prevent tearing in to minimised windows

### PR DESCRIPTION
**Pull request by owennw**

### CODE REVIEW

File: src/window-service.js [hunk 1]
```diff
@@ -94,26 +94,34 @@
...
+                        this.otherInstance = mainWindow;
+                        result = true;
+                    }
 
-            return false;
+                    deferred.resolve();
+                });
+            });
+
+            this.$q.all(promises).then(() => cb(result));
```
Comment (WPFerg) [inline src/window-service.js]: The native `Promise.race` might be a good thing to try, I don't think it'll make much difference aside from running time.

Comment (owennw) [inline src/window-service.js]: I'm a bit wary of using both `$q` and native promises (and then one day using `Q`). I think I'd rather stick to just the one. Also I don't see getting a window state to be a long wait... (famous last words).

Comment (WPFerg) [inline src/window-service.js]: That's fair enough, it makes sense for what little it would save.

**Comment by WPFerg** [comment]
Closing this, and using #370.

**Pull request closed without merging**